### ISSUE TITLE
SynthMorph shapes demo - change of loss functions

### ISSUE THREAD

**Issue by aungriah** [voxelmorph/voxelmorph#579]
Hello,

First of all, thank you for this great repository, it is extremely useful.
I have noticed that the SynthMorph demo google colab notebook has been modified: Before, a combination of Dice Score and Grad were used as loss functions, while the current notebook uses MSE and Grad loss. Since the MSE is not robust to different contrasts, I wanted to ask if this was done on purpose. In addition, the training script for SynthMorph still uses Dice and Grad...

Thank you very much for your time and help!

**Comment by adalca** [comment]
pinging @mu40

**Comment by mu40** [comment]
Well spotted, @aungriah! Recently, we were a bit unhappy with the performance of `vxm.losses.Dice` for deformable registration, as this implementation computes a mean of independent Dice terms across labels and thus disregards relative structure sizes. MSE is a great simple alternative that is sensitive to the structure size.

The framework remains contrast-agnostic, because SynthMorph optimizes a loss on probabilities (one-hot encoded label maps), rather than directly on the images. We will look into updating the training script soon.

**Comment by aungriah** [comment]
Thank you, that makes a lot of sense. Looking forward to seeing the effect on the registrations!
